### PR TITLE
Only configure :websocket-url if :websocket-host is provided

### PR DESCRIPTION
**Pull request by peterschwarz**
Given that `:websocket-host` defaults to `"localhost"`, when connected to a running figwheel instance from an external device (e.g. a tablet or another PC), the `:websocket-url` is configured to `"ws://localhost:3449/figwheel-ws"`.  The only way to get this to work is explicitly set the `:websocket-host` at a minimum.  If you're like me, you probably change wi-fi networks frequently, and therefore get assigned IP addresses frequently.  

Looking at source in the support subproject, `figwheel.client/config-defaults` properly handles creating the websocket url based on the javascript `location.host`.  So, automatically, it will connect to localhost if I've loaded the app via `http://localhost:3449`, or to my ip, if I've gone to `http://<my ip>:3449`.

Long story short, we don't need to default this when generating the `figwheel/connect.cljs`

### CODE REVIEW

**Comment by bhauman** [comment]
Peter I understand your reasoning here.  Having this auto-configuration is handy but only works as long as you are using the figwheel server to serve the assets.

And this is a tough one for me, but I think that having figwheel just work when you are serving your app from another server and even the filesystem by default is a big win. See the Quick Start. 

But I see this need as well. Maybe there is another configuration parameter to add instead? 
Or `:websocket-url-hook "example.core/websocket-url"` 
Or `:websocket-host :js-client-host`?
I'm going to have to think more on this.

**Comment by peterschwarz** [comment]
I hadn't thought of this use case.  I've been using it for Clojurescript-only projects at the moment, so I haven't set it up any other way than using figwheel.  If one is using other servers or the filesystem, does this require any different settings in the `project.clj`?

**Comment by bhauman** [comment]
Currently it does not require any change to the project.clj.  Because it has a knowledge of the port, etc.

**Comment by peterschwarz** [comment]
I've added `:js-client-host` (though, if you want to change the name, feel free - I'm not entirely sure it has quite the right description, but I'm at a loss at the moment for something better), and it seems like a good way to do it.  It leaves the original configuration, but then allows a way to configure things (or skip configuring, in this case), such that the ClojureScript side to make a guess at the websocket-url.

**Comment by bhauman** [comment]
This looks great thanks!
